presto-spi/src/main/java/com/facebook/presto/spi/resourceGroups/ResourceGroupId.java

```diff
@@ -29,6 +29,20 @@
 {
     private final List<String> segments;
 
+    public static ResourceGroupId resourceGroupId(String segmentedFullName)
```

Comment:
This method should be named something like `fromSegmentedName`. Also, it wasn't obvious that this is creating a recursive object. Add Javadoc for this method (very important since this is the SPI).

Another issue is this can be confused with the constructor that also takes a string. Add validation to the constructor that the name doesn't have dots.